### PR TITLE
[INLONG-10867][SDK] Transform support BETWEEN AND operator

### LINKED ISSUE #10867: [Feature][SDK] Transform support BETWEEN AND operator
### Description
- parent #10796 

- x BETWEEN n AND m
Check whether x within n and m.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ying-hua**
Fixes #10867

### Motivation

Add one BetweenOperator class  and test function class to support BETWEEN AND operator.
- x BETWEEN n AND m
Check whether x within n and m.
### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/operator/BetweenOperator.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * @param rowIndex
+     * @return
+     */
+    @SuppressWarnings("rawtypes")
+    @Override
+    public boolean check(SourceData sourceData, int rowIndex, Context context) {
+        return OperatorTools.compareValue((Comparable) this.left.parse(sourceData, rowIndex, context),
+                (Comparable) this.start.parse(sourceData, rowIndex, context)) >= 0
+                && OperatorTools.compareValue((Comparable) this.left.parse(sourceData, rowIndex, context),
+                        (Comparable) this.end.parse(sourceData, rowIndex, context)) <= 0;
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/operator/BetweenOperator.java:53]: It is repeat for the code "this.left.parse(sourceData, rowIndex, context)".

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/operator/OperatorTools.java [hunk 2]
```diff
@@ -165,6 +166,8 @@ public static ExpressionOperator buildOperator(Expression expr) {
             return new MinorThanOperator((MinorThan) expr);
         } else if (expr instanceof MinorThanEquals) {
             return new MinorThanEqualsOperator((MinorThanEquals) expr);
+        } else if (expr instanceof Between) {
+            return new BetweenOperator((Between) expr);
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/operator/OperatorTools.java:170]: Miss unit case.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/operator/BetweenOperator.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+package org.apache.inlong.sdk.transform.process.operator;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.operators.relational.Between;
+
+/**
+ * Between operator
```
Comment (aloyszhang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/operator/BetweenOperator.java:27]: Please add more details in the description

**Comment by luchunliang** [comment]
Please use annotation to resolve conflicting file.
Please use independent UT class file to avoid UT conflicting file.

**Comment by ying-hua** [comment]
> Please use annotation to resolve conflicting file. Please use independent UT class file to avoid UT conflicting file.
Done. I temporarily add TestBetweenAndOperator class, it will be synchronized with #11011 later.

**Review by aloyszhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**